### PR TITLE
Use serial_terminal for 'orphaned_packages_check'

### PR DESCRIPTION
**Pull request by rfan1**
As discussed in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/16704
Use serial_terminal is a better solution

- Related ticket: https://progress.opensuse.org/issues/126578
- Verification run: https://openqa.opensuse.org/tests/3197878

### CODE REVIEW

File: tests/console/orphaned_packages_check.pm [hunk 1]
```diff
@@ -55,21 +55,21 @@ sub compare_orphans_lists {
...
     # Save the orphaned packages list to one log file and upload the log, so QA can use this log to report bug
     # Filter out zypper warning messages and release or skelcd packages
-
-    script_run(
-q[zypper --quiet packages --orphaned | awk -F'|' 'intable && NR>2 {print $3} /^-/ { intable=1 }' | grep -v '\(release-DVD\|release-dvd\|openSUSE-release\|skelcd\)' > /tmp/orphaned.log],
+    my @orphans = split(
+        '\n',
+        script_output
+q[zypper --quiet packages --orphaned | tee -a /tmp/orphaned.log | awk -F'|' 'intable && NR>2 {print $3} /^-/ { intable=1 }' | grep -v '\(release-DVD\|release-dvd\|openSUSE-release\|skelcd\)'],
+        proceed_on_failure => 1,
```
Comment (grisu48) [inline tests/console/orphaned_packages_check.pm:70]: why is `proceed_on_failure` needed?

File: tests/console/orphaned_packages_check.pm [hunk 2]
```diff
@@ -55,21 +55,21 @@ sub compare_orphans_lists {
...
     zypper_call('in curl') if (script_run('rpm -qi curl') == 1);
     # Orphans are also expected in JeOS without SDK module (jeos-firstboot, jeos-license and live-langset-data)
     # Save the orphaned packages list to one log file and upload the log, so QA can use this log to report bug
     # Filter out zypper warning messages and release or skelcd packages
-
-    script_run(
-q[zypper --quiet packages --orphaned | awk -F'|' 'intable && NR>2 {print $3} /^-/ { intable=1 }' | grep -v '\(release-DVD\|release-dvd\|openSUSE-release\|skelcd\)' > /tmp/orphaned.log],
+    my @orphans = split(
+        '\n',
+        script_output
```
Comment (Vogtinator) [inline tests/console/orphaned_packages_check.pm:68]: IMO this is too much nesting.

File: tests/console/orphaned_packages_check.pm (hunk 1 above)
Comment (rfan1) [reply tests/console/orphaned_packages_check.pm:70]: Let me remove it

File: tests/console/orphaned_packages_check.pm (hunk 2 above)
Comment (rfan1) [reply tests/console/orphaned_packages_check.pm:68]: You are correct!

**Comment by rfan1** [comment]
VR is done: https://openqa.opensuse.org/tests/3199115#

**Review by foursixnine — APPROVED** [review]

File: tests/console/orphaned_packages_check.pm (hunk 2 above)
Comment (Vogtinator) [reply tests/console/orphaned_packages_check.pm:67]: Then change it?

**Review by Vogtinator — CHANGES_REQUESTED** [review]
Too much nesting

File: tests/console/orphaned_packages_check.pm (hunk 2 above)
Comment (rfan1) [reply tests/console/orphaned_packages_check.pm:67]: I did, but I think it is still too much!! let me try to do more

Comment (rfan1) [reply tests/console/orphaned_packages_check.pm]: @Vogtinator https://openqa.opensuse.org/tests/3199317#details

**Review by Vogtinator — APPROVED** [review]

**Review by grisu48 — APPROVED** [review]
Nice and tidy, LGTM!

**Comment by DimStar77** [comment]
https://openqa.opensuse.org/tests/3199527#step/orphaned_packages_check/14 - related?

**Comment by DimStar77** [comment]
https://openqa.opensuse.org/tests/overview?result=failed&result=incomplete&result=timeout_exceeded&distri=microos&distri=opensuse&version=Tumbleweed&build=20230329&groupid=1

many more showing up

**Pull request merged**